quickwit-indexing/src/source/mod.rs

```diff
@@ -198,3 +198,25 @@ pub struct SourceConfig {
     pub source_type: String,
     pub params: serde_json::Value,
 }
+
+impl SourceConfig {
+    pub async fn check(&self) -> anyhow::Result<()> {
+        match &*self.source_type {
+            "file" => {
+                if let Some(path) = self.params.get("filepath").and_then(|v| v.as_str()) {
+                    if !std::path::Path::new(path).exists() {
+                        anyhow::bail!("Source `{}` does not exist", path)
+                    }
+                }
+                Ok(())
+            }
+            #[cfg(feature = "kafka")]
+            "kafka" => kafka_source::check(self.params.clone()),
+            "vec" => Ok(()),
+            x => {
```

Comment:
rename x for something more suggestive like `unrecognized_source_type`
Generally speaking we try to avoid 1 letter variable names.